developer-extension/src/panel/hooks/useEvents.ts

```diff
@@ -15,24 +15,43 @@ export interface EventFilters {
   query: string
 }
 
-export function useEvents() {
+export function useEvents(preserveEvents: boolean) {
   const [events, setEvents] = useState<StoredEvent[]>([])
   const [filters, setFilters] = useState<EventFilters>({
     sdk: ['rum', 'logs'],
     query: '',
   })
 
-  useEffect(
-    () =>
-      listenRequests((newEvents) => {
-        setEvents((oldEvents) =>
-          [...newEvents, ...oldEvents]
-            .sort((first: any, second: any) => second.date - first.date)
-            .slice(0, MAXIMUM_LOGGED_EVENTS)
-        )
-      }),
-    []
-  )
+  useEffect(() => {
+    const removeListener = listenRequests((newEvents) => {
+      setEvents((oldEvents) =>
+        [...newEvents, ...oldEvents]
+          .sort((first: any, second: any) => second.date - first.date)
+          .slice(0, MAXIMUM_LOGGED_EVENTS)
+      )
+    })
+
+    const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
+      if (details.transitionType === 'reload') setEvents([])
+    }
+
+    useEffect(() => {
+      if (!preserveEvents) {
+        const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
+          if (details.transitionType === 'reload') setEvents([])
+        }
+        chrome.webNavigation.onCommitted.addListener(clearCurrentEvents)
+        return () => {
+          chrome.webNavigation.onCommitted.removeListener(clearCurrentEvents)
+        }
+      }
+    }, [preserveEvents])
+
+    return () => {
+      removeListener()
+      chrome.webNavigation.onCommitted.removeListener(clearCurrentEvents)
+    }
+  }, [])
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I'm really not sure nested `useEffects` are allowed! Suggestion: use separate `useEffects`:

```suggestion
  useEffect(() => {
    return listenRequests((newEvents) => {
      setEvents((oldEvents) =>
        [...newEvents, ...oldEvents]
          .sort((first: any, second: any) => second.date - first.date)
          .slice(0, MAXIMUM_LOGGED_EVENTS)
      )
    })
  }, [])

  useEffect(() => {
    if (!preserveEvents) {
      const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
        if (details.transitionType === 'reload') setEvents([])
      }
      chrome.webNavigation.onCommitted.addListener(clearCurrentEvents)
      return () => {
        chrome.webNavigation.onCommitted.removeListener(clearCurrentEvents)
      }
    }
  }, [preserveEvents])
```